internal/stack/stack.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stack
+
+// Stack is a generic stack implementation.
+type Stack[T any] struct {
```

Comment:
Will the `Stack` handle concurrency in the future?

Reply:
yes, it's not thread-safe in this pr.